File: Practica1/punto.java
public class Punto {
    public float x;
    public float y;

    public Punto(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float[] coord_cartesianas() {
        return new float[]{this.x, this.y};
    }

    public float[] coord_polares() {
        float radio = (float) Math.sqrt(this.x * this.x + this.y * this.y);
        float angulo = (float) Math.toDegrees(Math.atan2(this.y, this.x));
        return new float[]{radio, angulo};
    }

    @Override
    public String toString() {
        return String.format("(%.2f, %.2f)", this.x, this.y);
    }

    public static void main(String[] args) {
        Punto p1 = new Punto(0, 3);
        System.out.println(p1);
        float[] cartesianas = p1.coord_cartesianas();
        System.out.println("Coordenadas Cartesianas: " + cartesianas[0] + ", " + cartesianas[1]);
        float[] polares = p1.coord_polares();
        System.out.println("Coordenadas Polares: Radio = " + polares[0] + ", Ángulo = " + polares[1]);
    }
}

File: Practica1/punto.py
import math

class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return self.x, self.y

    def coord_polares(self):
        radio = math.sqrt(self.x ** 2 + self.y ** 2)
        angulo = math.degrees(math.atan2(self.y, self.x))
        return radio, angulo

    def __str__(self):
        return "({:.2f},{:.2f})".format(self.x, self.y)

# Ejemplo de uso
p1 = Punto(0, 3)
print(p1)
x, y = p1.coord_cartesianas()
print(f'Coordenadas Cartesianas: {x}, {y}')
r, a = p1.coord_polares()
print(f'Coordenadas Polares: Radio = {r}, Ángulo = {a}')
